### PR TITLE
Rtmp to hls

### PR DESCRIPTION
**Pull request by varsill**
This PR introduces RTMP to HLS demo.
A h264 stream containing B-frames can be used with that demo.

### CODE REVIEW

File: rtmp_to_hls/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# RtmpToHls
```
Comment (daniel-jodlos) [inline rtmp_to_hls/README.md:1]: We should probably mention what it is right around here

File: rtmp_to_hls/lib/rtmp_to_hls/pipeline.ex [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      id: __MODULE__,
+      start: {__MODULE__, :start_and_play, []}
+    }
+  end
+
+  def start_and_play() do
+    {:ok, pid} = start_link(nil)
+    play(pid)
+    {:ok, pid}
+  end
```
Comment (daniel-jodlos) [inline rtmp_to_hls/lib/rtmp_to_hls/pipeline.ex:43]: Looks like we're on Core 0.9 or using deprecated functions. We should probably bump it up

File: rtmp_to_hls/README.md (hunk 1 above)
Comment (daniel-jodlos) [reply rtmp_to_hls/README.md:1]: Also, I think we also need usage instructions, maybe including tutorial on setting up OBS to stream to this server

**Review by daniel-jodlos — CHANGES_REQUESTED** [review]
Phoenix stuff is really bloated, but looks good in general. However, we need to work on the README

File: rtmp_to_hls/lib/rtmp_to_hls/pipeline.ex (hunk 2 above)
Comment (varsill) [reply rtmp_to_hls/lib/rtmp_to_hls/pipeline.ex]: done

File: rtmp_to_hls/README.md (hunk 1 above)
Comment (varsill) [reply rtmp_to_hls/README.md]: done

File: rtmp_to_hls/mix.exs [hunk 3]
```diff
@@ -49,7 +49,7 @@ defmodule RtmpToHls.MixProject do
       ## Membrane deps
       {:membrane_core, "~> 0.10"},
       {:membrane_rtmp_plugin, "~> 0.6"},
-      {:membrane_http_adaptive_stream_plugin, "~> 0.8"},
+      {:membrane_http_adaptive_stream_plugin, "~> 0.8"}
```
[the remark this answers is not in the document]
Comment (varsill) [reply rtmp_to_hls/mix.exs]: good idea, done

File: rtmp_to_hls/README.md [hunk 4]
```diff
@@ -1,18 +1,60 @@
...
+```
+
+The server will be waiting for RTMP stream on `localhost:9009`, and the client of the application will be available on `localhost:4000`.
+### Running OBS
+Here is an example on how to generate RTMP stream. It uses [OBS](https://obsproject.com).
+Once you have OBS installed, you can perform the following steps:
+1. Open the OBS application
+2. Open the `Settings` windows
+3. Go to the `Stream` tab and set the value in the `Server` field to: `rtmp://localhost:9009` (the address where the server is waiting for the stream)
+4. Optionally you can go to the `Output` tab, under the `Stream` tab, set the `Output mode` to `Advanced`, and change the h264 `Profile` to the desired one
```
[the remark this answers is not in the document]
Comment (varsill) [reply rtmp_to_hls/README.md]: done

File: rtmp_to_hls/README.md [hunk 5]
```diff
@@ -4,12 +4,12 @@ This demo is an application which is capable of receiving RTMP stream, convertin
 ## Possible use cases
 The application presented in this demo could be used in the following scenario.
 There is one person, so called "streamer", and multiple "viewers", who want to see the stream of multimedia sent by the streamer.
-The streamer sents multimedia with the use of RTMP as it is protocol supported by the streamer's streaming tool (i.e. OBS). Such a RTMP stream is then converted into HLS and published by the HTTP server, which is capable of handling many HTTP requests. The viewers can then play the multimedia stream by downloading the files from the server with the use of HTTP. Such a solution scales well since the streamer do not need to have a direct connection with each of the viewers.
+The streamer sends multimedia using RTMP, a protocol supported by popular streaming software (i.e. OBS). Such a RTMP stream is then converted into HLS and published by the HTTP server, which is capable of handling many HTTP requests. The viewers can then play the multimedia stream by downloading the files from the server with the use of HTTP. Such a solution scales well because the streamer doesn't have a direct connection with any of the viewers.
```
Comment (daniel-jodlos) [inline rtmp_to_hls/README.md:7]: > The viewers can then play the multimedia stream by downloading the files from the server with the use of HTTP.

I'm wondering if this line isn't a little misleading 🤔 That's not necessarily true, is it?

File: rtmp_to_hls/README.md [hunk 6]
```diff
@@ -38,17 +38,18 @@ Finally, you can start the phoenix server:
...
+### Exemplary stream generation with OBS
+You can send RTMP stream onto `localhost:9009` with your favorite streaming tool. However, below we present how to generate an RTMP stream with
+[OBS](https://obsproject.com).
 Once you have OBS installed, you can perform the following steps:
 1. Open the OBS application
 2. Open the `Settings` windows
 3. Go to the `Stream` tab and set the value in the `Server` field to: `rtmp://localhost:9009` (the address where the server is waiting for the stream)
-4. Optionally you can go to the `Output` tab, under the `Stream` tab, set the `Output mode` to `Advanced`, and change the h264 `Profile` to the desired one 
-5. Finally, you can go back to the main windows and start streaming with the `Start Streaming` button.
+4. Finally, you can go back to the main windows and start streaming with the `Start Streaming` button.
```
Comment (daniel-jodlos) [inline rtmp_to_hls/README.md:50]:
```suggestion
4. Finally, you can go back to the main window and start streaming with the `Start Streaming` button.
```

Comment (varsill) [reply rtmp_to_hls/README.md:50]: done

File: rtmp_to_hls/README.md (hunk 5 above)
Comment (varsill) [reply rtmp_to_hls/README.md]: I have rearranged it into: 'The viewers can then play the multimedia delivered with HTTP'

**Review by daniel-jodlos — APPROVED** [review]

**Pull request merged**